File: algorithms/implementation/p04_find_digits/find_digits.py
#!/bin/python3

import sys

def to_arr(num):
    arr = []
    while num != 0:
        arr.append(num % 10)
        num = int(num / 10)
    arr.reverse()
    return arr

t = int(input().strip())
for a0 in range(t):
    n = int(input().strip())
    arr = to_arr(n)
    num_div = 0
    for i in range(0, len(arr)):
        if (arr[i] != 0) and (n % arr[i] == 0):
            num_div += 1
    print(num_div)            

File: algorithms/implementation/p04_find_digits/find_digits.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {
    
    public static List<Integer> getArr(int num) {
        List<Integer> arr = new ArrayList<Integer>();
        while (num != 0) {
            arr.add(num % 10);
            num /= 10;
        }
        Collections.reverse(arr);
        return arr;
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int t = in.nextInt();
        for(int a0 = 0; a0 < t; a0++){
            int n = in.nextInt();
            List<Integer> arr = Solution.getArr(n);
            int numDiv = 0;
            for (int i = 0; i < arr.size(); i++) {
                if ((arr.get(i) != 0) && (n % arr.get(i) == 0)) {
                    numDiv += 1;
                }
            }
            System.out.println(numDiv);
        }
    }
}
